compiler_opt/rl/log_reader_test.py

```diff
@@ -59,31 +62,34 @@ def create_example(fname: str):
                 'type': 'float'
             }
         }))
-    f.write(nl)
-    f.write(json_to_bytes({'context': 'some_context'}))
-    f.write(nl)
-    f.write(json_to_bytes({'observation': 0}))
-    f.write(nl)
-    write_buff(f, t0_val, ctypes.c_float)
-    write_buff(f, t1_val, ctypes.c_int64)
-    f.write(nl)
-    f.write(json_to_bytes({'outcome': 0}))
-    f.write(nl)
-    write_buff(f, s, ctypes.c_float)
-    f.write(nl)
-
-    t0_val = [v + 1 for v in t0_val]
-    t1_val = [v + 1 for v in t1_val]
-    s[0] += 1
-
-    f.write(json_to_bytes({'observation': 1}))
-    f.write(nl)
-    write_buff(f, t0_val, ctypes.c_float)
-    write_buff(f, t1_val, ctypes.c_int64)
-    f.write(nl)
-    f.write(json_to_bytes({'outcome': 1}))
-    f.write(nl)
-    write_buff(f, s, ctypes.c_float)
+    for ctx_id in range(nr_contexts):
```

Comment:
this looks like assembly to me.... you need to add comments for each line and/or wrap in functions with descriptions

Reply:
yup - good point, fixed.